### ISSUE TITLE
Error opening on Excel

### ISSUE THREAD

**Issue by rafaelSallowicz** [mk-j/PHP_XLSXWriter#144]
I'm creating a file and when I open that file in Excel appear a message error.
"Excel could not open <file name>.xlsx.  Do you want to open a repair this workbook?"
Clicking 'open and repair' allows the file to be opened without any noticeable modifications and then show me the logs.

<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<recoveryLog xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main"><logFileName>Reparar resultado pararesultado (1)0.xml</logFileName><summary>Erros detectados no arquivo '/Users/rafaelsallowicz/Downloads/resultado (1).xlsx'</summary><repairedRecords summary="A lista a seguir é de reparos:"><repairedRecord>Registros Reparados: Informações sobre a célula de /xl/worksheets/sheet1.xml</repairedRecord></repairedRecords></recoveryLog>

I've tried to change my sheet name and others modifications with any success.

**Comment by zaartix** [comment]
i have same problem

**Comment by startas** [comment]
Well, usually this happens when php throws error or warning, and that error gets written to your excel file, and it gets corrupted, but opening it and going through all messages still opens the file and you can see the error message.

**Comment by premax** [comment]
In my case in happened because I've set the value of a cell to more than 32767 characters and that is a maximum for Excel 2007 format.

**Comment by VladGogolev** [comment]
As @startas said, the problem happens mostly when php throws error or warning.
Try to check errors. 
```
ini_set('display_startup_errors',1);
ini_set('display_errors',1);
error_reporting(-1);
``` 
In my case, there was a problem with external libs. They throws some minor warnings and XLSXWriter tried to write them to the file.

If you need to process some rows before writing, you can try to do it at the first page and send $data array (via $_SESSION, it works ok) to the script on the second page, which is without any libs except XLSXWriter.

**Comment by VladGogolev** [comment]
Also, try to clean output buffer.
Here's an example: 

```
**ob_clean();
ob_end_flush();**
header("Cache-Control: no-cache, must-revalidate");
header("Content-Type: application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
header('Content-disposition: attachment; filesize="'.filesize($filename).'"');
header('Content-disposition: attachment; filename='.basename($filename));
echo file_get_contents($filename);
exit;
```
You need to write ob_clean() and ob_end_flush right before you are sending headers.
That might help in most situations.

**Comment by ritterg** [comment]
I use PHP_XLSXWriter in a laravel project.
For me, this error was generated by the output of the Debugbar, which was added to the excel file.
Setting
`APP_DEBUG=false`
in the .env file or adding
`app('debugbar')->disable();`
to the export script helped

**Comment by BurninLeo** [comment]
I encountered the issue when the data did not match the format code. We had several numeric columns ("0" and "0.00"), but not only one header line but two. The second header line contained texts that caused the warning in Excel.

My solution was to inherit from XLSXWriter and create a modified version of `writeSheetRow` -> `writeSheetRowWithFormats` that allows for setting custom data formats (text, "@" in that case). Solved the problem.
